### ISSUE TITLE
False Positive for SC1003

### ISSUE THREAD

**Issue by gnzlbg** [koalaman/shellcheck#1490]
#### For bugs
- Rule Id: SC1003 
- My shellcheck version (`shellcheck --version` or "online"):  version: 0.6.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC1003)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env sh
touch foo
grep '[\d|\w|-]* \\' foo
```

#### Here's what shellcheck currently says:

```
Line 3:
grep '[\d|\w|-]* \\' foo
                  ^-- SC1003: Want to escape a single quote? echo 'This is how it'\''s done'.
```

#### Here's what I wanted or expected to see:

Nothing probably, I want to match a `\` in the grep. The wiki page states:

> ### Exceptions
>
> If you want your single quoted string to end in a backslash, you can rewrite as 'string'\\ or ignore this warning.

But writing `grep '[\d|\w|-]* '\\ foo` does not work.

**Comment by zackw** [comment]
The equivalent of

```
grep '[\d|\w|-]* \\' foo
```

with the trailing backslash outside the quotes, is

```
grep '[\d|\w|-]* '\\\\ foo
```

because you need to supply two backslashes in a row to grep.

I think this is another argument for suppressing the warning when the character after `\'` is a space (as suggested in <https://github.com/koalaman/shellcheck/issues/1548#issuecomment-483767340>).
